### ISSUE TITLE
Generate bundles for single countries/hierarchies

### ISSUE THREAD

**Issue by hannesj** [whosonfirst/whosonfirst#2]
Sorry if this is the wrong repo, but the project structure confuses me a bit :sweat: 

It would be nice if it were possible to download bundles for single countries or other hierarchies, similar to how you can download all of the files per place type currently.

**Comment by thisisaaronland** [comment]
Two things:

1) This is a fine place to report issues absent knowing which specific repository to use.
2) Other bundles are definitely on the list or, more specifically, we are thinking about a general purpose `whosonfirst-meta` repository to define lists of things (all the PLACETYPE in a REGION, and so on) in WOF that can then be easily "bundled".

Is there a particular place / set of records you're after in the meantime?

**Comment by hannesj** [comment]
Ok, Thanks

We are currently using Pelias as geocoder in our project and as it has transformed to using wof rather than quattroshapes in most places, we would like to do the same. We are only importing data for one country (Finland) and would like to keep continue doing that. Therefore we would need a bundle of the country-record for Finland, all its children (and probably its parents) for the whosonfirst-importer as well as the wof-admin-lookup (which is using wof pip service).

**Comment by thisisaaronland** [comment]
Okay, this would be a good and useful exercise for us to kick the tires on things. I definitely can't get to it today and I might be able to start in on it tomorrow.

The goal is for these kinds of requests to be able to addressed quickly but since this is the first one you might have to suffer me thrashing around for a bit, figuring things out :-)

If there is no news here in, say... a week feel free to poke me (gently) ?

**Comment by hannesj** [comment]
Thanks, we are also just starting to look into it, so no hurry.

**Comment by hannesj** [comment]
@thisisaaronland, sorry to ping you, as I know you are busy with things. Just wanted to ask if there has been any progress or if there is something we could help with?

**Comment by riordan** [comment]
@thisisaaronland would it be possible to bootstrap this with a country metafile?

Perhaps a single CSV that's made up of:
`wof:id`, `iso:country`, `wof:placetype`

It'd also help us with Pelias so we could split the administrative lookup for addresses more efficiently.

**Comment by thisisaaronland** [comment]
Yes, you could grep each of the meta files filtering on iso (FI) and
concatenate them in to a single file and then feed that to wof-clone-meta.
I think...

**Comment by vesameskanen** [comment]
Hi @thisisaaronland, any progress on this direction? Per country bundles, compatible with wof-admin-mapping, would be a great improvement.

**Comment by vesameskanen** [comment]
Would  it be possible to get some guidelines how to implement per country bundles ourselves? I don't know much about WOF project structure yet, so all information/help would be welcome.

**Comment by thisisaaronland** [comment]
Apologies for the very slow reply.

Probably the easiest way to do this would be to filter all of the wof-*-latest.csv files in the `meta` directory by the `wof_country` field (specifically "FI") in to a new CSV file.

From there you could use the new CSV file as the input to the wof-clone-metafiles [1] script, specifying the `dest` parameter to be a local file:/// URI.

Does that make sense?

[1] https://github.com/whosonfirst/go-whosonfirst-clone/blob/master/cmd/wof-clone-metafiles.go

**Comment by vesameskanen** [comment]
Thanks @thisisaaronland! I will try the method you suggested.

**Comment by vesameskanen** [comment]
Hi @thisisaaronland, our wof-admin mapping system now filters metafiles by 'FI' content and fetches data using wof-clone-metafiles. Everything seems to work fine. Needless to say, data loading time, wof-mapping speed and space consumption improved by several orders of magnitude! Many thanks for your valuable help.

**Comment by thisisaaronland** [comment]
Excellent! I am very pleased to hear this is the case and that the meta files are working we "designed".

Tangentially related is this, which might be of use / interest:

https://github.com/whosonfirst/go-whosonfirst-concordances/blob/master/utils/csv2sqlite

As the name of the repository suggests it was originally written for wrangling concordances but on my way home last night I started thinking it's probably something that would work with the meta files as well. I am hoping to do some testing later today...

**Comment by thisisaaronland** [comment]
BTW, the `csv2sqlite` stuff mentioned above has moved over here:

https://github.com/whosonfirst/whosonfirst-csv2sqlite
